testgrid/cmd/summarizer/BUILD.bazel

```diff
@@ -1,8 +1,11 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
 
 go_library(
   name = "go_default_library",
-  srcs = ["summary.go"],
+  srcs = [
+      "summarizer.go",
```

Comment:
Consider adding a `main.go` with a `main` function, or renaming `summarizer.go` and adding the `main` function there.  Things in a `cmd` subdirectory are expected to be invoked from the command line

Reply:
Good point. Thanks Aaron. My original thought was this is more of a library than a binary. I will discuss with Michelle.

Reply:
It's currently a library, but we'll be running it as a binary (either by itself, or combined with other post-update jobs like alerting).

Reply:
Does some design for how this fits together with the rest of the world exist somewhere? Throwing out a library with one function that performs a trivial transformation from one mysterious format into another mysterious format leaves many questions and few answers. What else is coming, and what does it do?

Reply:
Actually this library has zero functions - nothing is exported. What is this library supposed to be?

Reply:
Sorry, to clarify, currently TestGrid runs code like this as a library as part of post-update stuff. We SHOULD be running it independently instead, and having the update master call out to it to generate the summary when an update finishes.

I think this is one of the details glossed over when explaining the architecture, so no explanation atm. Probably good to add to the README. The gist is that TestGrid runs the update master to parse all the groups, then makes request to update servers to get results and create the state.proto. After an update completes, the update master runs post-update jobs (currently summarizer (creates summary.proto) and alerter (mails alerts for newly-failing tests)). Atm summarizer is a module imported into the code, but we're moving to being able to run it separately.

The 'mysterious format' part is something we should be fixing in the short-term; currently there's some stuff with protos and JSON that are slightly different, but should be brought into alignment and made explicit.

Reply:
Thanks for all the input and clarification. I will be adding some detail to the README based on Michelle's response.

Also, this module has more functions. I decided to break it down into smaller, independent pieces (commits) to make it easier for reviewer's to do review. There will be roughly two more PRs. One is to create a client object from the summary.proto for the frontend client to use. The second will be the ability to interact with storage and maybe local cache to save and restore a summary object.

Reply:
I feel like small PRs are nice in principle, but in the absence of a published overarching design they just make it hard to know why you're doing the things you're doing, which makes them impossible to meaningfully review.

Can you put together a document explaining where this is headed, what the parts will be, and why it's here?

Reply:
Sounds good, will add that to the summarizer README